wallets/core/src/namespaces/common/hooks/changeAccountSubscriber.ts

```diff
@@ -0,0 +1,112 @@
+import type {
+  Actions,
+  SubscriberCleanUp,
+} from '../../../hub/namespaces/types.js';
+import type { Subscriber } from '../../../mod.js';
+import type {
+  AutoImplementedActionsByRecommended,
+  SupportedProviderTypes,
+} from '../types.js';
+
+type ChangeAccountSubscriberBuilderConfig<
+  EventType,
+  ProviderAPI extends SupportedProviderTypes,
+  AddEventListenerReturnType
+> = {
+  getInstance: () => ProviderAPI;
+  format: (instance: ProviderAPI, event: EventType) => Promise<string[]>;
+  shouldItDisconnect: (event: EventType) => boolean;
+  addEventListener: (
+    instance: ProviderAPI,
+    callback: (event: EventType) => void
+  ) => AddEventListenerReturnType;
+  removeEventListener: (
+    instance: ProviderAPI,
+    callback: (event: EventType) => void,
+    addEventListenerReturn: AddEventListenerReturnType
+  ) => void;
+};
+export class ChangeAccountSubscriberBuilder<
+  EventType,
+  AddEventListenerReturnType,
+  ProviderAPI extends SupportedProviderTypes,
+  ActionsType extends Actions<ActionsType> &
+    Actions<AutoImplementedActionsByRecommended>
+> {
+  private config = {} as ChangeAccountSubscriberBuilderConfig<
```

Comment:
```suggestion
  private operations = {} as ChangeAccountSubscriberBuilderOperations<
```
I suggest to name this `operations`  as they are atomic units of work.